src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java

```diff
@@ -217,4 +223,135 @@ public final void testUpdateInstitution(final Vertx aVertx, final VertxTestConte
+        myScheduleStoreProxy.listJobs().onSuccess(jobList -> {
+            aContext.verify(() -> {
+                assertTrue(jobList != null);
+                assertTrue(jobList.size() >= 3);
+                assertTrue(jobList.get(0).getScheduleCronExpression().toString().equals(SAMPLE_CRON));
+            }).completeNow();
+        }).onFailure(aContext::failNow);
+    }
+
+    /**
+     * Tests deleting job from db.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @Test
+    public final void testDeleteJob(final Vertx aVertx, final VertxTestContext aContext)
+            throws AddressException, MalformedURLException, NumberParseException, ParseException {
+        final Job toDelete = TestUtils.getRandomJob();
+        myScheduleStoreProxy.addJob(toDelete).onSuccess(newID -> {
+            myScheduleStoreProxy.removeJob(newID).onSuccess(result -> {
+                myScheduleStoreProxy.getJob(newID).onFailure(details -> {
+                    final ServiceException error = (ServiceException) details;
+
+                    aContext.verify(() -> {
+                        assertEquals(Error.NOT_FOUND.ordinal(), error.failureCode());
+                        assertTrue(error.getMessage().contains(String.valueOf(newID)));
+                    }).completeNow();
+                }).onSuccess(select -> {
+                    aContext.failNow(LOGGER.getMessage(MessageCodes.PRL_014, newID));
+                });
+            }).onFailure(aContext::failNow);
+        }).onFailure(aContext::failNow);
+    }
+
+    /**
+     * Tests updating job in db.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @Test
+    public final void testUpdateJob(final Vertx aVertx, final VertxTestContext aContext)
+            throws AddressException, MalformedURLException, NumberParseException {
+        final int jobID = 1;
+        myScheduleStoreProxy.getJob(jobID).onSuccess(original -> {
+            try {
+                final Job modified = new Job(original.getInstitutionID(), new URL("http://new.url.com"),
+                        original.getSets().get(), original.getScheduleCronExpression(), ZonedDateTime.now());
+                myScheduleStoreProxy.updateJob(jobID, modified).onSuccess(result -> {
+                    myScheduleStoreProxy.getJob(jobID).onSuccess(updated -> {
+                        aContext.verify(() -> {
+                            assertTrue(updated.getInstitutionID() == original.getInstitutionID());
+                            assertTrue(updated.getRepositoryBaseURL().equals(modified.getRepositoryBaseURL()));
+                        }).completeNow();
+                    }).onFailure(aContext::failNow);
+                }).onFailure(aContext::failNow);
+            } catch (MalformedURLException details) {
+                aContext.failNow(details);
+            }
+        }).onFailure(aContext::failNow);
+    }
+
```

Comment:
Re: my comment about the update query failing if the institutionID was changed, there should be a test verifying that behavior.

Reply:
Added test for failed update event,